sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -437,10 +438,6 @@ RestClient createClient() throws IOException {
               return requestConfigBuilder;
             }
           });
-      if (getSocketAndRetryTimeout() != null) {
-        restClientBuilder.setMaxRetryTimeoutMillis(getSocketAndRetryTimeout());
-      }
-
```

Comment:
You remove it for all ES versions. So, this change is not backward compatible. Maybe a simple if ?